Correctness/ArmSpec.lean

```diff
@@ -18,4 +18,20 @@ theorem arm_run (n : Nat) (s : ArmState) :
     simp only [Sys.run, Sys.next, run]
     rw [h]
 
+-- (TODO) Move to Arm/BitVec.lean?
+/-- Unexpander to represent bitvector literals in hexadecimal -- this overrides
+  the unexpander in the Lean BitVec library. -/
+@[app_unexpander BitVec.ofNat] def unexpandBitVecOfNatToHex : Lean.PrettyPrinter.Unexpander
+  | `($(_) $n:num $i:num) =>
+      let i' := i.getNat
+      let n' := n.getNat
+      let trimmed_hex := -- Remove leading zeroes...
+        String.dropWhile (BitVec.ofNat n' i').toHex
+                         (fun c => c = '0')
+                         -- ... but keep one if the literal is all zeros.
+      let trimmed_hex := if trimmed_hex.isEmpty then "0" else trimmed_hex
+      let bv := Lean.Syntax.mkNumLit s!"0x{trimmed_hex}#{n'}"
+      `($bv:num)
+  | _ => throw ()
+
```

Comment:
One reason why I'm hesitant re. this unexpander is that register indices also appear in hex then. E.g., `StateField.GPR 0xc#5` and `StateField.SFP 0xd#5`, which is inconvenient.

Maybe we need more directed unexpanders...

Reply:
Something I've been thinking about: it might be nice to introduce a simple abbreviation for addresses and raw instructions
```
abbrev Address := BitVec 64

abbrev RawInstr := BitVec 32
```

Then, we could somehow define the unexpanders to apply just for those types, not for all bitvectors

Reply:
Yes, that's a good idea, regardless of the unexpander issue.

Note that it also helps to see machine components' values as hex; for instance, consider:
```
r (StateField.GPR 1#5) s = 18446744073709551614#64
```
vs.
```
r (StateField.GPR 1#5) s = 0xfffffffffffffffe#64
```